db/version_set.cc

```diff
@@ -3221,32 +3231,36 @@ void VersionStorageInfo::UpdateFilesByCompactionPri(
     if (num > temp.size()) {
       num = temp.size();
     }
-    switch (ioptions.compaction_pri) {
-      case kByCompensatedSize:
-        std::partial_sort(temp.begin(), temp.begin() + num, temp.end(),
-                          CompareCompensatedSizeDescending);
-        break;
-      case kOldestLargestSeqFirst:
-        std::sort(temp.begin(), temp.end(),
-                  [](const Fsize& f1, const Fsize& f2) -> bool {
-                    return f1.file->fd.largest_seqno <
-                           f2.file->fd.largest_seqno;
-                  });
-        break;
-      case kOldestSmallestSeqFirst:
-        std::sort(temp.begin(), temp.end(),
-                  [](const Fsize& f1, const Fsize& f2) -> bool {
-                    return f1.file->fd.smallest_seqno <
-                           f2.file->fd.smallest_seqno;
-                  });
-        break;
-      case kMinOverlappingRatio:
-        SortFileByOverlappingRatio(*internal_comparator_, files_[level],
-                                   files_[level + 1], ioptions.clock, level,
-                                   num_non_empty_levels_, options.ttl, &temp);
-        break;
-      default:
-        assert(false);
+    // We don't pick last level files based on compaction priority,
+    // so we don't need to do the sorting.
+    if (level != num_levels() - 1) {
```

Comment:
the `level` should never be bottommost:
https://github.com/facebook/rocksdb/blob/48ce44240c398850a752d13d1a27c9849ae6fad2/db/version_set.cc#L3216-L3217

Reply:
Ahh. Got it.